### PR TITLE
Add exporter context env vars to jmp shell and env_with_metadata() helper

### LINKED ISSUE #53: exporter context under a shell
When using the jmp shell, with j commands, or with the python libraries, sometimes it's necessary to get some context of the connected exporter

the env() util should return exporter details as well, or have an additional method, and jmp shell should provide information of the connected exporter on connection (i.e. name, labels, etc...)

For example we could have env_with_metadata() that provides client, exporter_metadata.

```python
with env_with_metadata() as client, exporter_metadata:
    print(exporter_metadata.name)
    print(exporter_metadata.labels)

```

be able to retrieve:
   * board name
   * labels
   * any other important metadata
 

via @eballetbo

### PR DESCRIPTION
**Pull request by mangelajo**
## Summary

- Sets `JMP_EXPORTER`, `JMP_LEASE`, and `JMP_EXPORTER_LABELS` environment variables when launching a `jmp shell` with a remote lease, so scripts and commands running inside the shell can discover the exporter context.
- Adds `env_with_metadata()` / `env_with_metadata_async()` context managers and `ExporterMetadata` dataclass that let Python code retrieve the exporter name, labels, and lease ID from within a shell session.
- Fetches exporter labels from the controller during lease acquisition and stores them on the `Lease` object.

### Usage

```python
from jumpstarter.common.utils import env_with_metadata

with env_with_metadata() as (client, metadata):
    print(metadata.name)    # e.g. "my-board-exporter"
    print(metadata.labels)  # e.g. {"board": "rpi4", "location": "lab-1"}
    print(metadata.lease)   # e.g. "lease-abc123"
```

Or directly from the shell environment:

```bash
echo $JMP_EXPORTER        # my-board-exporter
echo $JMP_LEASE           # lease-abc123
echo $JMP_EXPORTER_LABELS # board=rpi4,location=lab-1
```

Closes https://github.com/jumpstarter-dev/jumpstarter/issues/53

## Test plan

- [x] Existing tests pass (162 jumpstarter + 39 jumpstarter-cli)
- [x] New test: `test_launch_shell_sets_lease_env` verifies env vars are set when a lease is provided
- [x] New test: `test_exporter_metadata_from_env` verifies parsing with all env vars set
- [x] New test: `test_exporter_metadata_from_env_empty` verifies graceful behavior with no env vars
- [x] New test: `test_exporter_metadata_from_env_labels_with_equals_in_value` verifies label values containing `=`
- [x] Lint passes cleanly

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by evakhoni** [comment]
> Or directly from the shell environment:
echo $JMP_EXPORTER        # my-board-exporter
echo $JMP_LEASE           # lease-abc123
echo $JMP_EXPORTER_LABELS # board=rpi4,location=lab-1

I wonder what would happen with complex selectors. are we fetching the selector we leased with, or the labels of the exporter that ended up matching that selector?

File: python/packages/jumpstarter/jumpstarter/client/lease.py [hunk 1]
```diff
@@ -138,6 +139,14 @@ async def request_async(self):
...
         return await self._acquire()
 
+    async def _fetch_exporter_labels(self):
+        """Fetch the exporter's labels after lease acquisition."""
+        try:
+            exporter = await self.svc.GetExporter(name=self.exporter_name)
+            self.exporter_labels = exporter.labels
+        except Exception:
+            logger.debug("Could not fetch labels for exporter %s", self.exporter_name)
+
```
[the remark this answers is not in the document]
Comment (evakhoni) [reply python/packages/jumpstarter/jumpstarter/client/lease.py:211]: wonder in which cases can it be that we're unable to fetch exporter labels? if it's something unlikely that can happen only in case of an actual issue with the exporter or something, then we may want to be more verbose than debug :thinking:

**Review by raballew — DISMISSED** [review]
The label serialization (`key1=val1,key2=val2`) will break on Kubernetes label values that contain commas -- e.g. `env=dev,staging` would get parsed as two separate labels. JSON would handle this cleanly, or at least reject/document the limitation.

Also the test `test_launch_shell_sets_lease_env` only checks `exit_code == 0` but doesn't actually verify the env vars show up in the output. And `env_with_metadata()` / `env_with_metadata_async()` and `_fetch_exporter_labels` don't have tests yet.

File: python/packages/jumpstarter/jumpstarter/common/utils_test.py [hunk 2]
```diff
@@ -22,3 +24,52 @@ def test_launch_shell(tmp_path, monkeypatch):
...
+    )
+    exit_code = launch_shell(
+        host=str(tmp_path / "test.sock"),
+        context="my-exporter",
+        allow=["*"],
+        unsafe=False,
+        use_profiles=False,
+        lease=lease,
+    )
+    assert exit_code == 0
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/common/utils_test.py:57]: This test proves the shell launched but doesn't actually check that `JMP_EXPORTER`, `JMP_LEASE`, and `JMP_EXPORTER_LABELS` were set to the right values. Since `SHELL` is set to `env`, you could capture stdout and assert on the expected env var values - or use a small wrapper script that writes specific vars to a temp file for verification.

File: python/packages/jumpstarter/jumpstarter/client/lease.py [hunk 3]
```diff
@@ -179,6 +181,15 @@ async def request_async(self):
...
         return await self._acquire()
 
+    async def _fetch_exporter_labels(self):
+        """Fetch the exporter's labels after lease acquisition."""
+        try:
+            exporter = await self.svc.GetExporter(name=self.exporter_name)
+            self.exporter_labels = exporter.labels
+        except Exception as e:
+            self.exporter_labels = {}
+            logger.warning("Could not fetch labels for exporter %s: %s", self.exporter_name, e)
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/client/lease.py:210]: This has two code paths (success + exception fallback) but neither is directly unit-tested. The happy path only runs during full lease acquisition against a real controller. Would be nice to have a couple of unit tests mocking `self.svc.GetExporter` - one for success, one for the exception fallback.

File: python/packages/jumpstarter/jumpstarter/common/utils.py [hunk 4]
```diff
@@ -84,6 +92,43 @@ def _run_process(
...
+        JMP_DRIVERS_ALLOW: "UNSAFE" if unsafe else ",".join(allow),
+        "_JMP_SUPPRESS_DRIVER_WARNINGS": "1",  # Already warned during client initialization
+    }
+    if insecure:
+        env = env | {JMP_GRPC_INSECURE: "1"}
+    if passphrase:
+        env = env | {JMP_GRPC_PASSPHRASE: passphrase}
+    if lease is not None:
+        env.update(_lease_env_vars(lease))
+    return env
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/common/utils.py:129]: New helper with a decent parameter matrix (insecure, passphrase, lease combos) but it's only tested indirectly through `launch_shell()`. Direct unit tests for this function would make it easier to catch regressions - especially edge cases like `passphrase=""` (falsy, so `JMP_GRPC_PASSPHRASE` won't be set - is that intentional?).

_AI-generated, human reviewed_

File: python/packages/jumpstarter/jumpstarter/utils/env.py [hunk 5]
```diff
@@ -1,12 +1,46 @@
...
+        name = os.environ.get(JMP_EXPORTER, "")
+        lease = os.environ.get(JMP_LEASE) or None
+
+        labels: dict[str, str] = {}
+        raw_labels = os.environ.get(JMP_EXPORTER_LABELS, "")
+        if raw_labels:
+            for pair in raw_labels.split(","):
+                if "=" in pair:
+                    k, v = pair.split("=", 1)
+                    labels[k] = v
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/utils/env.py]: Minor: if `JMP_EXPORTER_LABELS` contains something like `=value`, this creates `{"": "value"}` since `split("=", 1)` yields `["", "value"]`. Unlikely with k8s labels but a quick `if k:` guard before the assignment would be cleaner.

**Review by raballew — CHANGES_REQUESTED** [review]

File: python/packages/jumpstarter/jumpstarter/utils/env.py (hunk 5 above)
Comment (mangelajo) [reply python/packages/jumpstarter/jumpstarter/utils/env.py]: The labels really come from k8s, they cannot have that form :) but an "if k:" really won't hurt, I am buying :)

**Comment by mangelajo** [comment]
@ambient-code please rebase this

File: python/packages/jumpstarter/jumpstarter/utils/env.py [hunk 6]
```diff
@@ -54,3 +102,23 @@ def env():
...
+
+        with env_with_metadata() as (client, metadata):
+            print(metadata.name)
+            print(metadata.labels)
+            print(metadata.lease)
+    """
+    with start_blocking_portal() as portal:
+        with ExitStack() as stack:
+            with portal.wrap_async_context_manager(env_with_metadata_async(portal, stack)) as result:
+                yield result
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/utils/env.py:124]: [LOW] No direct tests for `env_with_metadata()` / `env_with_metadata_async()` context managers.

These are thin wrappers -- `env_with_metadata_async` wraps `env_async` and adds `ExporterMetadata.from_env()`. The underlying components are individually tested, and testing these wrappers would require mocking the full gRPC client setup. Acceptable as-is given the well-tested components, but consider adding an integration test if practical.

_AI-generated, human reviewed_

File: python/packages/jumpstarter/jumpstarter/common/utils_test.py [hunk 7]
```diff
@@ -22,3 +27,174 @@ def test_launch_shell(tmp_path, monkeypatch):
...
+    output = env_output.read_text()
+    assert "JMP_EXPORTER=my-exporter" in output
+    assert "JMP_LEASE=lease-123" in output
+    assert "board=rpi4" in output
+    assert "location=lab-1" in output
+
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/common/utils_test.py:63]: [LOW] Test uses substring matching instead of verifying the full label value.

The test checks `"board=rpi4" in output` and `"location=lab-1" in output` separately. This would pass even if the labels were unsorted, had extra spaces, or had a different format.

Suggested fix: assert the complete value instead:
```python
assert "JMP_EXPORTER_LABELS=board=rpi4,location=lab-1" in output
```

_AI-generated, human reviewed_

File: python/packages/jumpstarter/jumpstarter/utils/env.py [hunk 8]
```diff
@@ -1,12 +1,47 @@
...
+)
+
+
+@dataclass(frozen=True)
+class ExporterMetadata:
+    """Metadata about the exporter connected to the current shell session."""
+
+    name: str
+    labels: dict[str, str] = field(default_factory=dict)
+    lease: str | None = None
```
Comment (raballew) [inline python/packages/jumpstarter/jumpstarter/utils/env.py:27]: [LOW] Minor inconsistency in default values: `name` defaults to `""` (empty string) while `lease` defaults to `None`.

This creates different null-checking patterns: `if meta.name:` vs `if meta.lease is not None:`. Both work in practice and this appears to be a deliberate choice (name is always "present" as a string, lease is truly optional). Could be documented in the docstring for clarity.

_AI-generated, human reviewed_

**Review by raballew — APPROVED** [review]
